File: app/src/main/java/com/soumya/wwdablu/zomatobuddy/common/SearchTypes.java
package com.soumya.wwdablu.zomatobuddy.common;

import android.support.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class SearchTypes {

    @Retention(RetentionPolicy.SOURCE)
    @StringDef({
            SEARCH_DINE_OUT,
            SEARCH_DELIVERY,
            SEARCH_TAKE_AWAY
    })
    public @interface SearchType{}

    public static final String SEARCH_DINE_OUT = "2";
    public static final String SEARCH_DELIVERY = "1";
    public static final String SEARCH_TAKE_AWAY = "5";
}

File: app/src/main/java/com/soumya/wwdablu/hungry/common/SearchTypes.kt
package com.soumya.wwdablu.hungry.common

import androidx.annotation.StringDef

object SearchTypes {
    const val SEARCH_DINE_OUT: String = "2"
    const val SEARCH_DELIVERY: String = "1"
    const val SEARCH_TAKE_AWAY: String = "5"
    const val SEARCH_FAVOURITE: String = "999"

    @kotlin.annotation.Retention(AnnotationRetention.SOURCE)
    @StringDef(SEARCH_DINE_OUT, SEARCH_DELIVERY, SEARCH_TAKE_AWAY, SEARCH_FAVOURITE)
    annotation class SearchType
}